react/Sandbox.tsx

```diff
@@ -51,110 +64,131 @@ function init (options: IframeOptions) {
         }
         const [clean] = value.split(';');
         (window as any).__cookie = `${clean}; ` + (window as any).__cookie
-        window.parent.postMessage({type: options.cookieEventType, value}, '*')
+        window.parent.postMessage({type: cookieEventType, value}, '*')
       },
     })
   }
 
+  // Add styles
+  if (styles && Array.isArray(styles)) {
+    styles.forEach((style: StyleContainer) => {
+      if (style.href) {
+        const link = document.createElement('link')
+        link.href = style.href
+        link.type = 'text/css'
+        link.rel = 'stylesheet'
+        document.head.appendChild(link)
+      } else if (style.rules) {
+        const sheet = document.createElement('style')
+        sheet.innerHTML = style.rules
+        document.head.appendChild(sheet)
+      }
+    })
+  }
+
   window.addEventListener('message', function(event: MessageEvent) {
-    const {styles, props, content, cookie} = event.data
+    const {props} = event.data
```

Comment:
You're assuming that all postMessage events are from sandbox. You should check if it's the sandbox event. Otherwise the property `props` might be `undefined`.

Reply:
True that!